### PR TITLE
[docs] cleanup and update Emotion setup

### PR DESCRIPTION
**Pull request by Simek**
# Why

Current setup of [`emotion`](https://emotion.sh/) contains mix of old and new packages. 

The new version simplifies the configuration and work out-of-the box with Next.js and SSR. 

Let's clean this up and updated to latest major version.

# How

Summary of changes:
* package name change/migration according to:
  * https://emotion.sh/docs/emotion-11#package-renaming
  * `@emotion/react` is an extended `@emotion/css` (with `Global` component etc.)
* according to docs`@emotion/babel-preset-css-prop` includes `@emotion/babel-plugin` (so there is no need to add `@emotion/babel-plugin` separately):
  * https://emotion.sh/docs/@emotion/babel-preset-css-prop#usage
  * unfortunately, it fails at `build`, but adding a new types declaration to `next-evn.d.ts` solves this issue, this can be removed after upgrade to TS 4.1+ (https://emotion.sh/docs/emotion-11#css-prop-types)
* SSR and Next.js are supported out-of-the box, with default configuration
  * https://emotion.sh/docs/ssr#nextjs (check out "Note")
  * https://emotion.sh/docs/ssr#default-approach
  * only limitation is that default setup do not support `nth-` CSS selectors, which I cannot find in current docs styles, so it should work fine in docs case

> Note: I have also tried to mimic previous server setup first (without reading more Emotion docs) and it looks like that new `@emotion/server` package requires framework agnostic `@emotion/css` package, and having only `@emotion/react` in dependencies is not enough.

# Test Plan

Docs website compiling and running on local environment. Successful `yarn build`.

I was not able to spot any visual regression, after applying the changes, in `dev` and `export-server` modes.

It also look likes (from failed snapshots), that upgrading fixed the issue of duplicated class names.

@bbarthec @brentvatne @jonsamp

### CODE REVIEW

**Review by bbarthec — APPROVED** [review]
Looks promising 😊 
Keeping tooling/dependencies up-to-up for the win 🚀 🥳

File: docs/pages/_document.tsx [hunk 1]
```diff
@@ -21,18 +20,9 @@ const DynamicLoadAnalytics = dynamic<any>(() =>
 export default class MyDocument extends Document<{ css?: string }> {
   static async getInitialProps(ctx: DocumentContext) {
```
Comment (axeldelafosse) [inline docs/pages/_document.tsx:21]: We can probably get rid of this

**Review by jonsamp — APPROVED** [review]
Very nice 🎉

**Comment by bbarthec** [comment]
@Simek, fix the conflicting file and I'll merge it 😉

File: docs/pages/_document.tsx (hunk 1 above)
Comment (Simek) [reply docs/pages/_document.tsx:15]: Thank you for the hint, I will look at this and prepare next PR if needed!

**Review by brentvatne — APPROVED** [review]

**Pull request merged**